src/lib/comms/include/sol-http-server.h

```diff
@@ -278,6 +278,35 @@ sol_http_response_set_sse_headers(struct sol_http_response *response)
 }
 
 /**
+ * @brief Progressive server response configuration.
+ * @struct sol_http_server_progressive_config
+ * @see sol_http_server_send_progressive_response()
+ */
+struct sol_http_server_progressive_config {
+    /**
+     * @brief Callback used to inform that a struct @ref sol_blob was sent
+     * @param data The user data
+     * @param progressive The progressive response
+     * @param blob The blob that was transfered
+     * @note The blob will unref for you automatically
+     */
+    void (*on_feed)(void *data, const struct sol_http_progressive_response *progressive, struct sol_blob *blob);
+    /**
+     * @brief Callback used to inform that the client has closed the connection.
+     *
+     * @param data The user data
+     * @param progressive The progressive response
+     */
+    void (*on_close)(void *data, const struct sol_http_progressive_response *progressive);
+    const void *user_data; /*<< User data to @c on_feed and @c on_close*/
+    /**
+     * The output buffer size - 0 means unlimited data.
+     * @see sol_http_progressive_response_feed()
+     */
+    size_t max_bytes;
```

Comment:
do you want to rename this to `tx_size` now or later when the other PR documenting the design pattern goes in?